### DISCUSSION TITLE
Nightscout on Google Cloud with an existing noip.com hostname

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2420] [General]
There are reports of installation failing.  Let's use this thread to gather info.

**Comment by mattcorzine** [comment]
- I was originally having trouble with the No-IP setup portion. I started with an existing noip.com account (no hostnames existed - old account that I haven't used in a while).
- When I ran Step 2 - "noip.com association" I would be asked for username/password. After supplying this I believe there was an error but it disappeared so quickly that I couldn't capture it.
- Searching through comments on Facebook, one user mentioned "you have to create a new hostname, and add the free SSL cert to the host, then wait for No-IP to verify the cert (can take an hour), and ONLY then do the No-IP configuration in the NS installer on Google Cloud".
- I logged into noip.com and created a hostname (it seems like this is performed when someone is creating a new No-IP account).
- Afterwards I executed Step 2 - "noip.com association" again. This time the installer asked me to choose a hostname that I wanted to attach to and a few other basic questions, which I left at default except the API_Secret and email address.
- The installer appeared to create some SSL certificate, and now when browsing to that address, my NS page loads and is a valid HTTPS site (no warnings or errors).

My new NS site is getting closer to being ready - I hope my stumble and documentation of what worked for me helps someone else.
Thanks @Navid200!

**Reply by mattcorzine** [reply]
@Navid200 just wanted to update that my NS site is fully up and running on Google Cloud. Now I have to figure out how to move tconnectsync off of heroku.

This isn't related to this topic but not sure if it's worth a new topic or not:
The only other 'gottchas' I had were with the NS variables. First, the Bridge, Mongo, and other custom variables didn't migrate automatically, so I had to type them manually. Then I found the Enable variable had been populated with some standard options but didn't include 'bridge'. Once I had the variables figured out then it was up and running.

**Reply by Navid200** [reply]
You can open new discussion threads as you wish.  I am talking "discussions" not "issues".

We are working on improving the variable access, edit, view, transfer to make it more user friendly.

**Reply by tinerfeno** [reply]
Hello!
Today doing another installation from the beginning and having created a previous No-IP account for a previous installation, I created a new No-Ip Account and I was able to obtain this error. As @mattcorzine has said it was impossible to advance in second step of the installation because there were an error of user/password unknow, despite new No-IP account was fully working. 
To try to solve this issue, I reopened another Google Chrome session with other user (other gmail address) that was not used before to create an account of No-IP, entered No-IP with the new Account created before and then I run the second step of the installation, and it works. 
I suppose No-Ip accounts are linked to gmail users of Google Chrome, and when you open Google Chrome as a new user, the association is relinked to the new user and then there is no problem in the second step of the installation.
I hope it helps. 
Regards. Daniel

**Reply by Navid200** [reply]
We are bringing FreeDNS as an alternative to noip.com.
It will be released soon.

**Comment by Rodolphe77** [comment]
Hi there,

I follow the step-by-step installation process, but it appears I have No-IP issue.

My hostname is not locked and I am asked to install a SSL certificate I don't know how.

On Step 2 - I always receive an error message in red color : "Challenge failed for domain XXXXXX.horto.org" and later (firewall issue).

Can you detail more how we have to set up our domain in No-IP.com ?

Thank you all for your hard work.

**Reply by Navid200** [reply]
I can think of two options.

My preferred option is to wait for the next release.  The next release will use freeDNS instead of noip.com.  I expect to submit my new updates tomorrow.  At that point, after it is tested, it will be released.  I cannot promise exactly when that will happen.  But, I hope it will not be long.

The second option I can think of is to create a new noip.com account using a different email address.  And in this new account, only create a single hostname to be used for Nightscout on Google Cloud.  Then, the install process will complete.

As soon as the new release is out, I will post here and in the facebook group.

**Reply by phjund** [reply]
Hello,
I have a ssh problem in the deployment of Nightscout on Google.
I have a first error in step 5 "pre-installation of Nightscout". After entering the 4th time my passphrase I have the following error:
ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [255].
I nevertheless was able to continue the installation until the end but at the 2nd step of the Xdrip procedure "noip.com association" I get the following message:
********
Performing the following challenges:
http-01 challenge for XXX.ddns.net
Waiting for verification...
Challenge failed for domain XXX.ddns.net
http-01 challenge for XXX.net
Cleaning up challenges
Some challenges have failed.

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: XXX.net
   Type:   connection
   Detail: XX.XXX.128.128: Fetching
   http://XXX.ddns.net/.well-known/acme-challenge/-MIVy2s_K9K-0DzWrxbhpXiNSojolYyN_-AXsZaIeHY:
   Timeout during connect (likely firewall problem)

   To fix these errors, please make sure that your domain name was
   entered correctly and the DNS A/AAAA record(s) for that domain
   contain(s) the right IP address. Additionally, please check that
   your computer has a publicly routable IP address and that no
   firewalls are preventing the server from communicating with the
   client. If you're using the webroot plugin, you should also verify
   that you are serving files from the webroot path you provided.
****************
I checked that the IP address on No-ip is correct and that it points to the VM on Google. But I guess the web server of the VM is not correctly installed but I don't have the knowledge to debug this.
I should add that this is my 2nd trial to install Nightscout on Google but even though I restarted from scratch it didn't help.
Could you please help me to solve this problem ?
Thanking you in advance,
Phil.

**Reply by Navid200** [reply]
@phjund Did you create a new noip.com account and created a hostname to use for Nightscout on Google Cloud?
Or, did you have an existing noip.com account and created a second hostname to use in your existing account?

**Reply by phjund** [reply]
Thanks for your quick answer.
I created a new noip.com account and created a new hostname to use for Nightscout on Google Cloud.

**Reply by Navid200** [reply]
> I have a first error in step 5 "pre-installation of Nightscout". 

What do you mean by step 5?

**Reply by phjund** [reply]
I followed the installation procedure given here :
https://digital-diabetes.com/2022/10/31/deploy-nightscout-easily-on-google-cloud-for-free/
Step 5 corresponds to "Pre-Installation"...

**Reply by Navid200** [reply]
That's a copy.
The only guide I can supports is the original guide here:
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud

You have two options:
1- Follow our instructions, and I will go out of my way to support you.
2- Follow a copy, and if you need help, ask for help from the person who chose to copy instead of providing a link to the original.

**Reply by phjund** [reply]
I'm sorry ! I wasn't aware that the instructions I followed were not the originals....At the end the person who wrote the copy gives a link that points here in case of troubles thus I thought he/she is an acquaintance.
I will try to do the installation a third time using the link you gave me but if it is similar to the copy, than the result will probably be the same...
Again I'm sorry for my mistake and in any case I thank you for all the work you are doing for making our life a little bit easier !

**Reply by Navid200** [reply]
He contacted us and said that he wanted to support individuals who did not speak English.
I thanked him and told him that was wonderful.

The next thing I found out was that he had changed his guide to English!
I told him that was not right.

**Reply by phjund** [reply]
English is not my native language (and I don't speak Spanish !) so I appreciate your help and I'm surprised that such behaviors can happen in our "community"...

**Reply by Navid200** [reply]
Please follow our guide and let me know if it doesn't work.

**Reply by phjund** [reply]
OK will do...thanks again.

**Reply by phjund** [reply]
Hello again,
I followed your instructions and indeed I don't have any ssh problems anymore...but I have another (probably basic) problem.
My NS webpage is working fine but I don't know how to feed the data of my Dexcom G6 into Nightscout ! In your tutorial the last step concerns using Xdrip but I don't have Xdrip and I don't know how to tell Nightscout to read directly my Dexcom readings: is this possible ? Is Xdrip mandatory ?

**Reply by Navid200** [reply]
I am glad you were able to complete the installation.

The question you are asking is a general question that is related to any host you use for Nightscout.
Is this the first time you are setting up Nightscout?
If yes, you can ask your question here:  https://www.facebook.com/groups/cgminthecloud

If you have set up Nightscout before, for example on Heroku, you can use the exact same settings you used there.

**Reply by phjund** [reply]
In fact your way of installing NS on Google is much simpler than the other one: thanks again !
I have already a Heroku instance of Nightscout that is still working fine but I don't know how I managed to do this !!! With your tutorial I selected "4 Copy data from another Nightscout" thinking that this would do the trick, but it only copies the past results and doesn't update the data.
I think with Heroku I could choose where I want to download the data from (Dexcom, Xdrip,..etc..) but here I was never asked such a question and I don't know where I can enter this information.

**Reply by Rodolphe77** [reply]
> I can think of two options.
> 
> My preferred option is to wait for the next release. The next release will use freeDNS instead of noip.com. I expect to submit my new updates tomorrow. At that point, after it is tested, it will be released. I cannot promise exactly when that will happen. But, I hope it will not be long.
> 
> The second option I can think of is to create a new noip.com account using a different email address. And in this new account, only create a single hostname to be used for Nightscout on Google Cloud. Then, the install process will complete.
> 
> As soon as the new release is out, I will post here and in the facebook group.

Hi Navid200, in fact the issue was from the Firewall within Google Cloud : Default-SSH need to  be edited and checked the tipbox that allows all port.
Since I have changed that (thanks to a FB thread (Tomasz Tloczek)), I can now access my nightscout website. Cheers

**Reply by tzachi-dar** [reply]
You really only need to allow http and https. No need to open all ports.

**Comment by tzachi-dar** [comment]
In order to copy the data from NS please use the script:
 sudo /srv/nightscout-vps/GetHerokuVars.sh
 It should show you your current heroku settings.

**Reply by phjund** [reply]
Thank you for your response.
I can have access to the Heroku settings, no problem, but I don't know how to enter these settings in the Google hosted Nightscout...
Maybe with "3 Edit Nightscout variables" but when I look at this I see only:
export API_SECRET="XXX"
export ENABLE="careportal food boluscalc bwp cob bgi pump openaps rawbg iob upbat cage sage basal"
export AUTH_DEFAULT_ROLES="denied"
export PUMP_FIELDS="reservoir battery clock"
export DEVICESTATUS_ADVANCED="true"

**Reply by Navid200** [reply]
You need to follow the instructions here and add any additional variables to the defaults you need:
https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables

We don't have a utility to transfer the variables from another Nightscout to our Nightscout installation.
At least, not yet.

As you have discovered, we only have a utility that let's you copy readings and treatments from another Nightscout to ours.

**Reply by phjund** [reply]
I have already done what you have suggested and added "bridge" in ENABLE because this is necessary to read from Dexcom Share but it doesn't help....I have nowhere to tell Nightscout it should read the data from Dexcom and give the BRIDGE informations that I have from Heroku.

**Reply by tzachi-dar** [reply]
Can you try running
```
sudo cp heroku_vars.txt /etc/nsconfig
sudo reboot
```
This will copy all your variables from heroku to the new site.
In any case, hopefully, today I'll send a PR that will allow you to configure the variables with a modern UI.

**Reply by Navid200** [reply]
Look at your Nightscout variables on Heroku.  Have you entered them all here?

**Reply by phjund** [reply]
Thank you for your answers.
@tzachi-dar: I opened a terminal (option 11) and wrote your command but the answer was:
phjund@nightscout:~$ sudo cp heroku_vars.txt /etc/nsconfig
cp: cannot stat 'heroku_vars.txt': No such file or directory
which makes sense because I don't have the requested file on Google.
Sorry but I don't know what a PR is...
@Navid200: I have all the variables of the ENABLE key on Heroku, here. But of course I don't have all the other keys like
BRIDGE_SERVER, BRIDGE_PASSWORD, BRIDGE_USER_NAME, etc...there are a lot of keys that don't appear in the "Nightscout Variables" on Google...should I copy all of them with the corresponding values ??

**Reply by tzachi-dar** [reply]
Please run the script that I wrote before

**Reply by Navid200** [reply]
@phjund tzschi-dar is a senior designer here.  He knows a LOT more than I do.  
So, please follow his advice.

**Reply by phjund** [reply]
Of course I have no doubt about the qualifications of tzachi-dar I simply forgot to run the script.
When I do run it I get the following message:
 ›   Error: Your account has MFA enabled; API requests using basic authentication with email and password are 
 ›   not supported. Please generate an authorization token for API access. 
 ›
 ›   Error ID: vaas_enrolled
 ▸    Invalid credentials provided.
I guess this is because there is a 2 step identification when you want to log in to Heroku....

**Reply by jonprogers** [reply]
@phjund get your api key from your heroku account settings and use that for your password in the script rather than your account password - I think that's what made it get round the MFA issue for me...

**Reply by phjund** [reply]
@jonprogers Thanks for the tip...but unfortunately it doesn't work me :-(

**Reply by jonprogers** [reply]
@phjund that's a shame - the other thing I tried but didn't think worked was adding a .netrc file as instructed in this post
https://stackoverflow.com/questions/67707487/heroku-login-multi-factor-authentication-mfa-required-error-without-opening-a-b
maybe both are needed - you can tell I'm no expert here!! Good luck

**Reply by phjund** [reply]
OK now it works !
I copied one by one the conf. variables of the Heroku Nightscout into the /etc/nsconfig file and that made the trick. 
Without the help of @Navid200 and @tzachi-dar I wouldn't have been able to find the solution: thank you again for your help and support.
I think it should not be too difficult for you to include the 3 keys for Dexcom in the initial setup procedure of NS on Google which would make your solution straightforward even for a guy like me who is not a computer expert and who uses Dexcom as primary source for NS.
